test/ninja/mkdir_function_command.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+#
+# Copyright The SCons Foundation
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY
+# KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+
+import os
+
+import TestSCons
+from TestCmd import IS_WINDOWS
+
+test = TestSCons.TestSCons()
+
+try:
+    import ninja
+except ImportError:
+    test.skip_test("Could not find module in python")
+
+_python_ = TestSCons._python_
+_exe = TestSCons._exe
+
+ninja_bin = os.path.abspath(os.path.join(
+    ninja.__file__,
+    os.pardir,
+    'data',
+    'bin',
+    'ninja' + _exe))
+
+test.write('SConstruct', """
+SetOption('experimental','ninja')
+DefaultEnvironment(tools=[])
+env = Environment()
+env.Tool('ninja')
+env.Dir('test0')
+env.Dir('test1/test2')
+env.Dir('test3/test4')
+env.Command('test5', 'test3/test4', Mkdir('$TARGET'))
+env.Command('test6/test7', ['test1/test2', 'test0'], Mkdir('$TARGET'))
+""")
+
+# generate simple build
+test.run(stdout=None)
+test.must_contain_all_lines(test.stdout(), ['Generating: build.ninja'])
+test.must_contain_all(test.stdout(), 'Executing:')
+test.must_contain_single_instance_of(test.stdout(), ["scons: Building targets ..."])
+test.must_contain_all(test.stdout(), 'ninja%(_exe)s -f' % locals())
+test.must_exist([
+    test.workpath('test0'),
+    test.workpath('test1/test2'),
+    test.workpath('test3/test4'),
+    test.workpath('test5'),
+    test.workpath('test6/test7')
+])
+
+# clean build and ninja files
+test.run(arguments='-c', stdout=None)
+test.must_contain_all_lines(test.stdout(), [
+    'Removed build.ninja'])
+os.rmdir('test0')
+os.rmdir('test1/test2')
+os.rmdir('test1')
+os.rmdir('test3/test4')
+os.rmdir('test3')
+os.rmdir('test5')
+os.rmdir('test6/test7')
+os.rmdir('test6')
+
+test.must_not_exist([
```

Comment:
You're verifying that os.rmdir()'s working here?

Reply:
I suppose it is a little silly. I was verifying that the work dir is in a clean state, because the proceeding tests expect that.